pgml-extension/src/bindings/transformers.py

```diff
@@ -110,22 +110,38 @@ def transform(task, args, inputs):
     return json.dumps(pipe(inputs, **args), cls=NumpyJSONEncoder)
 
 
-def embed(transformer, text, kwargs):
+def embed(transformer, inputs, kwargs):
+    
+    try:
+        inputs = json.loads(inputs)
+    except json.decoder.JSONDecodeError:
+        pass      
+        
     kwargs = json.loads(kwargs)
     ensure_device(kwargs)
     instructor = transformer.startswith("hkunlp/instructor")
+    
     if instructor:
         klass = INSTRUCTOR
-        text = [[kwargs.pop("instruction"), text]]
+        if isinstance(inputs, str):
+            inputs = [[kwargs.pop("instruction"), inputs]]
+            
+        else:
+            texts_with_instructions = []
+            instruction = kwargs.pop("instruction")
+            for text in inputs:
+                texts_with_instructions.append([instruction, text])
+                
+            inputs = texts_with_instructions
     else:
         klass = SentenceTransformer
 
     if transformer not in __cache_sentence_transformer_by_name:
         __cache_sentence_transformer_by_name[transformer] = klass(transformer)
     model = __cache_sentence_transformer_by_name[transformer]
 
-    result = model.encode(text, **kwargs)
-    if instructor:
+    result = model.encode(inputs, **kwargs)
+    if instructor and len(result) == 1:
         result = result[0]
```

Comment:
I think we also need to handle the multi instructor case, right? Or does instructor always just return an appropriate single dimension array?